### PR TITLE
mungebot: Ensure milestone munger updates warning on label changes

### PR DESCRIPTION
**Pull request by marun**
Previously the milestone munger would only update warning notifications on the warning interval to avoid generating the notification on each pass.  This could result in confusion when label changes rendered the warning out-of-date.  This change ensures the warning is kept current with an issue's label state.

### CODE REVIEW

File: mungegithub/mungers/milestone-maintainer.go [hunk 1]
```diff
@@ -324,12 +318,25 @@ func labelsIncompleteState(issue *githubapi.Issue, notification *c.Notification,
...
 	}
 }
 
+// warningIsCurrent indicates whether the given notification
+// represents is an up-to-date warning.  The notification is the
+// previous warning from the munger.  The message is the warning
+// generated from the current labels on the issue.
+func warningIsCurrent(notification *c.Notification, message string, commentCreatedAt *time.Time, warningInterval time.Duration) bool {
+	hasNotification := (notification != nil && notification.Arguments == milestoneLabelsIncomplete)
+	return hasNotification && notification.Context == message && time.Since(*commentCreatedAt) < warningInterval
```
Comment (cjwagner) [inline mungegithub/mungers/milestone-maintainer.go]: Check that `commentCreatedAt` is not nil.

Comment (marun) [reply mungegithub/mungers/milestone-maintainer.go]: It's not necessary in this case.  If notification is not nil, comment is not nil, and commentCreatedAt will not be nil.

Comment (cjwagner) [reply mungegithub/mungers/milestone-maintainer.go]: Sorry, I don't follow. `c.ParseNotification` can return a non-nil notification even if the comment it is created from has a nil value for `CreatedAt`. So {notification not nil} does imply {comment not nil} but does not imply {commentCreatedAt not nil}.

Comment (marun) [reply mungegithub/mungers/milestone-maintainer.go]: Done

**Pull request merged**